Makefile

```diff
@@ -313,10 +315,33 @@ endif
 ### end of "build" target ###
 #############################
 
-
+std:
+##  Cache std/Cargo.toml
+	cp $(RUST_SOURCE)/library/std/Cargo.toml $(ROOT_DIR)/std-cargo.toml
+##	Remove the last line of std/Cargo.toml
+	@$(HEAD) -n -1 $(RUST_SOURCE)/library/std/Cargo.toml > $(ROOT_DIR)/temp-std-cargo.toml
+	@mv $(ROOT_DIR)/temp-std-cargo.toml $(RUST_SOURCE)/library/std/Cargo.toml
+##	Add the correct dependency path, since the dependency path in std/Cargo.toml assumes that std is
+##	being built using -Zbuild-std.
+	@echo "theseus-shim = { path = \"../../../../shim\", features = ['rustc-dep-of-std'] }" >> $(RUST_SOURCE)/library/std/Cargo.toml
+
+##	Build the compiler
+	@cd $(RUST_SOURCE) && RUSTFLAGS="" CARGOFLAGS="" ./x.py build library --stage 1
+	@cd $(ROOT_DIR)
+
+##	Remove the last line of std/Cargo.toml
+	@$(HEAD) -n -1 $(RUST_SOURCE)/library/std/Cargo.toml > $(ROOT_DIR)/temp-std-cargo.toml
+	@mv $(ROOT_DIR)/temp-std-cargo.toml $(RUST_SOURCE)/library/std/Cargo.toml
+## Restore previous std/Cargo.toml
+	mv $(ROOT_DIR)/std-cargo.toml $(RUST_SOURCE)/library/std/Cargo.toml
+
+##	Add .cargo/config.toml
+	@mkdir -p $(ROOT_DIR)/.cargo
+	@echo "[build]" > $(ROOT_DIR)/.cargo/config.toml
+	@echo "rustc = \"$(RUST_SOURCE)/build/$(HOST_TRIPLE)/stage1/bin/rustc\"" >> $(ROOT_DIR)/.cargo/config.toml
 
 ## This target invokes the actual Rust build process via `cargo`.
-cargo:
+cargo: std
```

Comment:
bump: i think we decided that `std` should not be built by default (?). Same goes for the `clippy` and `doc` targets, i think  those ideally shouldn't depend on std.

Reply:
You're reviewing an older commit.